pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,441 @@
+# coding=utf-8
+"""Test the API endpoints `OSTree`_ `repositories`_.
+
+This module assumes that the tests in
+:mod:`pulp_smash.tests.platform.api_v2.test_repository` hold true. The
+following trees of assumptions are explored in this module::
+
+    It is possible to create an OSTree repo with feed (CreateTestCase).
+    ├── When valid feed and branch are given repo synces successfully without
+    │   reported errors (CreateValidFeedTestCase).
+    ├── When Invalid feed or branch is given repo sync fails with errors
+    │   reported (SyncInvalidFeedTestCase).
+    └── It is possible to sync repository, copy its content to a second
+        repository, update repository metadata, sync again, add distributor
+        to the first repository and publish it and access published content
+        (PublishTestCase).
+
+    It is possible to create repository without feed (CreateTestCase).
+    └── Running sync on this repository fails with errors reported
+        (SyncWithoutFeedTestCase).
+
+.. _OSTree:
+    http://pulp-ostree.readthedocs.org/en/latest/
+.. _repositories:
+   http://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/repo/cud.html
+"""
+from __future__ import unicode_literals
+
+from itertools import chain
+from urlparse import urljoin
+from unittest2 import TestCase
+from packaging.version import Version
```

Comment:
Please split standard library imports from third-party imports.

`urlparse.urljoin` only exists on Python 2. A `try: ... except ImportError: ...` would fit the bill.